Tests/NIOPosixTests/ByteBufferTest.swift

```diff
@@ -1746,6 +1746,30 @@ class ByteBufferTest: XCTestCase {
         XCTAssertEqual(0xcc, slice.getInteger(at: 0, as: UInt8.self))
         XCTAssertEqual(0xdd, slice.getInteger(at: slice.writerIndex - 1, as: UInt8.self))
     }
+
+    func testSliceOfMassiveBufferWithAdvancedReaderIndexIsOk() throws {
+        // We only want to run this test on 64-bit systems: 32-bit systems can't allocate buffers
+        // large enough to run this test safely.
+        guard MemoryLayout<Int>.size >= 8 else {
+            return
```

Comment:
Nit: Can we use `XCTSkip` here instead of just returning
```suggestion
            throw XCTSkip("This test is only supported on 64-bit systems")
```